src/EnergyPlus/OutputProcessor.cc

```diff
@@ -5091,311 +3593,317 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
-                                                                                  state.dataGlobal->CalendarYear);
-        }
-
-        for (TimeStepType thisTimeStepType : {TimeStepType::Zone, TimeStepType::System}) { // Zone, HVAC
-            for (int Loop = 1; Loop <= op->NumOfRVariable; ++Loop) {
-                if (op->RVariableTypes(Loop).timeStepType != thisTimeStepType) continue;
-                auto &rVar = op->RVariableTypes(Loop).VarPtr;
-                // Update meters on the TimeStep  (Zone)
-                if (rVar.MeterArrayPtr != 0 && !state.dataOutputProcessor->MeterValue.empty()) {
-                    Real64 TimeStepValue = rVar.TSValue * rVar.ZoneMult * rVar.ZoneListMult;
-                    for (int i = 1; i <= op->VarMeterArrays(rVar.MeterArrayPtr).NumOnMeters; i++) {
-                        int index = op->VarMeterArrays(rVar.MeterArrayPtr).OnMeters(i);
-                        state.dataOutputProcessor->MeterValue(index) += TimeStepValue;
+        if (rf->timeSeriesEnabled()) {
+            if (!rf->freqTSData[(int)ReportFreq::TimeStep].variablesScanned()) {
+                rf->initializeTSDataFrame(ReportFreq::TimeStep, op->outVars);
+            }
+            rf->freqTSData[(int)ReportFreq::TimeStep].newRow(state.dataEnvrn->Month,
+                                                               state.dataEnvrn->DayOfMonth,
+                                                               state.dataGlobal->HourOfDay,
+                                                               op->TimeValue[(int)TimeStepType::Zone].CurMinute,
+                                                               state.dataGlobal->CalendarYear);
+        }
+
+
+        // Update meters on the TimeStep  (Zone)
+        if (op->meterValues.capacity() > 0) {
+            for (int iMeter = 0; iMeter < (int)op->meters.size(); ++iMeter) {
+                auto *meter = op->meters[iMeter];
+                if (meter->type == MeterType::Normal || meter->type == MeterType::Custom) {
+                    for (int srcVarNum : meter->srcVarNums) { 
+                       auto *var = op->outVars[srcVarNum];
+                       // Separate the Zone variables from the HVAC variables using TimeStepType
+                       if (var->timeStepType != TimeStepType::Zone && var->timeStepType != TimeStepType::System) continue;
+                       // Add to the total all of the appropriate variables, make sure to use var->TSValue and not *var->Which
+                       op->meterValues[iMeter] += var->TSValue * var->ZoneMult * var->ZoneListMult;
                     }
-                    for (int i = 1; i <= op->VarMeterArrays(rVar.MeterArrayPtr).NumOnCustomMeters; i++) {
-                        int index = op->VarMeterArrays(rVar.MeterArrayPtr).OnCustomMeters(i);
-                        state.dataOutputProcessor->MeterValue(index) += TimeStepValue;
+                } else if (meter->type == MeterType::CustomDec) { 
```

Comment:
New way of updating CustomDec meters. Sum all variables associated with "master" base meter, then subtract all variables associated with individual decrement variables and meters.